akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/Behaviors.scala

```diff
@@ -222,8 +222,8 @@ object Behaviors {
   final class Supervise[T] private[akka] (val wrapped: Behavior[T]) extends AnyVal {
 
     /** Specify the [[SupervisorStrategy]] to be invoked when the wrapped behavior throws. */
-    def onFailure[Thr <: Throwable: ClassTag](strategy: SupervisorStrategy): Behavior[T] = {
-      val tag = classTag[Thr]
+    def onFailure[Thr <: Throwable](strategy: SupervisorStrategy)(
+        implicit tag: ClassTag[Thr] = ThrowableClassTag): Behavior[T] = {
```

Comment:
What happened here?  I'd expect implicit lookup to default this to `ClassTag[Throwable]`.

Reply:
https://github.com/lampepfl/dotty/issues/9586

```scala
scala> def f[T <: Throwable : reflect.ClassTag](x: Int): Option[T] = ???
def f
  [T <: Throwable](x: Int)(implicit evidence$1: reflect.ClassTag[T]): Option[T]

scala> f(1)
1 |f(1)
  |    ^
  |    No ClassTag available for Nothing
```

The default is `ThrowableClassTag` because the the next line replaces `ClassTag.Nothing` with `ThrowableClassTag`.